Source/GoBot/GoBot.swift

```diff
@@ -79,6 +79,8 @@ class GoBot: Bot {
     }
     
     func login(network: NetworkKey, hmacKey: HMACKey?, secret: Secret, completion: @escaping ErrorCompletion) {
+        
+        // is this saying we need to be the main thread to login? -rabble
```

Comment:
Yes, this crashes the app if this function is not called in the main thread

Reply:
@cryptix shouldn't the rest of the code be enclosed in self.queue.async {
}
Maybe there is a good reason to not be that way

Reply:
@martindsq can't really answer that question but we can try running this in the background, too.